metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -378,34 +397,39 @@ void build_pred_succ(const std::string &graph_fname,
         std::vector<bool> pred_boundary_buf;
 
         for (uint64_t i = start; i < std::min(start + BS, graph.num_nodes() + 1); ++i) {
-            BOSS::edge_index boss_idx = graph.kmer_to_boss_index(i);
-            if (!dummy[boss_idx]) {
-                BOSS::edge_index next = boss.fwd(boss_idx);
-                assert(next);
-                if (!dummy[next]) {
-                    while (rd_succ.size() && !rd_succ[next]) {
-                        next--;
-                        assert(!boss.get_last(next));
-                    }
-                    succ_buf.push_back(to_row(graph.boss_to_kmer_index(next)));
+            bool skip_succ = false, skip_all = false;
+            if (succinct) { // Legacy code for DBGSuccinct
+                BOSS::edge_index boss_idx = from_graph_index(graph, i);
+                if((*dummy)[boss_idx]) {
+                    skip_all = true;
+                } else {
+                    skip_succ = (*dummy)[succinct->get_boss().fwd(boss_idx)];
```

Comment:
I'm trying to understand what is the intention of this check in the context of `row_diff_successor` function. Generally, do we want `row_diff_successor` to be well-defined for all nodes of the graph and return a graph node? If yes, then this convention would be violated for terminal nodes. How do we want to handle it? Should `row_diff_successor` return an `std::optional` on nodes that don't have a successor? Which nodes should it be in the terms of `graph::DeBruijnGraph`, if any?